src/main/java/core/basesyntax/FigureSupplierImpl.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplierImpl implements FigureSupplier {
```

Comment:
The class name `FigureSupplierImpl` uses a verb, which is not recommended. Consider renaming it to something like `FigureSupplierImplementation` or another noun-based name.